RELEASE.md

```diff
@@ -6,30 +6,46 @@ Official Falcosecurity rules releases are automated using GitHub Actions. Each r
+
 **NOTE:** *The versioning guidelines also apply to any versioned ruleset not maintained in this repository (such as the ones in [falcosecurity/plugins](https://github.com/falcosecurity/plugins)), including the ones distributed by third parties. These are best practices that guarantee the correct behavior of Falco when updating a given ruleset to a new version.*
 
 - `z` _(patch number)_ is incremented when you make backward-compatible changes. In this case, the ruleset can be updated in a given Falco without needing to update Falco, its plugins, or its configuration. Examples:
+    - Decrementing `required_engine_version`
+    - Decrementing plugin version requirement in `required_plugin_versions`
+    - Adding a plugin version requirement alternative in `required_plugin_versions`
+    - Removing a plugin version requirement with all its alternatives in `required_plugin_versions`
     - Enabling at default one or more rules that used to be disabled
     - Adding or removing items for one or more lists
-    - Adding or removing exceptions for one or more Falco rules (without needing a `required_engine_version`)
-    - Changing the condition for one or more rules or macros by still preserving their logical security scope (e.g. making them less noisy)
+    - Adding one or more tags to a rule
+    - Increasing the priority of a rule
+    - Changing the output fields of a rule (without increasing `required_engine_version`)
+    - Adding or removing exceptions for one or more Falco rules (without increasing `required_engine_version`)
+    - Changing the condition for one or more rules by still preserving their logical security scope (e.g. making them less noisy, matching more events than before)
 - `y` _(minor number)_ is incremented when you add functionality in a backward-compatible manner. In order to be accepted, the new ruleset may mandate updating the version of Falco, changing its configuration, or updating/installing one or more plugins. Examples:
-    - Incrementing the `required_engine_version` number
+    - Incrementing `required_engine_version`
     - Incrementing the `required_plugin_versions` version requirement for one or more plugin
-    - Adding a new plugin version requirement in `required_plugin_versions`
+    - Adding a new plugin version requirement (or an alternative) in `required_plugin_versions`
     - Adding one or more lists, macros, or rules
+    - Adapting the ruleset to a Falco engine version introducing *backward-compatible* changes in the expected ruleset language definitions or file format
```

Comment:
This is not clear to me. Do you mean without changing the `required_engine_version`? Could you provide an example please?

Reply:
> Adapting the ruleset to a Falco engine version introducing *backward-compatible* changes in the expected ruleset language definitions or file format

For example, where in the past we introduced the `append` or `exceptions` features in the YAML format of Falco rules, I would have expected only the minor version of the ruleset to be bumped. This is because migrating to the new feature didn't require changes in previously-existing rulesets if not simply "adding" new things inside it. This would have not broken expectations about all the previosly-defined lists, macros, rules, and how they are resolved by the engine.

> Adapting the ruleset to a Falco engine version introducing *backward-incompatible* changes in the expected ruleset language definitions or file format

Think about the case in which we fix some bugs (such as making the string escaping parsing sound), or that we alter the way the Falco engine resolves rules and their dependencies. Indeed, this would be a breaking change from the standpoint of the Falco engine, but it will also be a breaking change for the ruleset as well (thus requiring bumping the major version). For example, this could break assumptions about when a rule triggers or doesn't, what could be the expected alert, or potentially can change the `append` semantics on which users may rely on when appending their definitions to the default rules.

Reply:
> > Adapting the ruleset to a Falco engine version introducing _backward-compatible_ changes in the expected ruleset language definitions or file format
> 
> For example, where in the past we introduced the `append` or `exceptions` features in the YAML format of Falco rules, I would have expected only the minor version of the ruleset to be bumped. This is because migrating to the new feature didn't require changes in previously-existing rulesets if not simply "adding" new things inside it. This would have not broken expectations about all the previosly-defined lists, macros, rules, and how they are resolved by the engine.

This makes sense if we don't bump `required_engine_version` in such a case.

Rationale: since there's no way to know from `required_engine_version` if a backward-incompatible change has been introduced, we can only assume that the change might not be backward-compatible.

(it would have been different if `required_engine_version` was a semver-like version number. Are we assuming it will be? :thinking: )

Reply:
> (it would have been different if required_engine_version was a semver-like version number. Are we assuming it will be? 🤔 )

I assume it *may* be in the future. I don't want to exclude this possibility. Let me try to rephrase it better.